opendrift/models/basemodel.py

```diff
@@ -3294,39 +3299,52 @@ def get_map_background(self, map, background, time=None):
 
         return map_x, map_y, scalar, u_component, v_component
 
-    def get_density_array(self, pixelsize_m, weight=None):
+    def get_density_array(self, pixelsize_m, density_proj=None, llcrnrlon=None,llcrnrlat=None,urcrnrlon=None,urcrnrlat=None, weight=None):
+        '''
+        compute a particle concentration map from particle positions
+        Use user defined projection (density_proj=<proj4_string>)
+        or create a lon/lat grid (density_proj=None)
+        '''
+        times = self.get_time_array()[0]
         lon = self.get_property('lon')[0]
         lat = self.get_property('lat')[0]
-        times = self.get_time_array()[0]
-        deltalat = pixelsize_m/111000.0  # m to degrees
-        deltalon = deltalat/np.cos(np.radians((np.nanmin(lat) +
-                                               np.nanmax(lat))/2))
-        lat_array = np.arange(np.nanmin(lat)-deltalat,
-                              np.nanmax(lat)+deltalat, deltalat)
-        lon_array = np.arange(np.nanmin(lon)-deltalat,
-                              np.nanmax(lon)+deltalon, deltalon)
-        bins=(lon_array, lat_array)
+        if density_proj is None: # add default projection with equal-area property 
+            density_proj = pyproj.Proj('+proj=moll +ellps=WGS84 +lon_0=0.0')
+
+        # create a grid in the specified projection
+        x,y = density_proj(lon, lat)
```

Comment:
I discovered that the new write_netcdf_density_map() writes a grid which is much too small (e.g. 3x4 instead of 150x200 pixels). The reason (or part of the reason) seems to be that density_proj(lon, lat) returns not x,y in meters, but lon, lat, and apparently in radians! This might be an explanation:
http://pyproj4.github.io/pyproj/stable/gotchas.html#proj-not-a-generic-latitude-longitude-to-projection-converter